cloudinit/util.py

```diff
@@ -542,6 +542,11 @@ def is_ipv4(instr):
     return len(toks) == 4
 
 
+@lru_cache()
+def is_BSD():
+    return system_info()['variant'].endswith("bsd")
```

Comment:
are we sure this is lowercase everywhere, or is it lowercased elsewhere?

Reply:
It's either `freebsd`, `netbsd` or `openbsd`.

Reply:
So for the Linux world, we have `system_info()['platform'] == "linux"` and `system_info()['variant'] == "ubuntu"` (for example). Should we align the BSD world a little more along these hierarchical lines, so `system_info()['platform'] == "bsd"` (and `"variant"` remains as currently defined)?

As far as I can tell, "platform" is only used in a few places (this is all instances of the string `"system"`):

```
$ astpath "//Str[@s='system']" -d cloudinit
cloudinit/util.py:654  >	        'system': platform.system(),
cloudinit/util.py:660  >	    system = info['system'].lower()
cloudinit/stages.py:79   >	            system_config = self._extract_cfg('system')
cloudinit/stages.py:103  >	        elif restriction == 'system':
cloudinit/config/cc_chef.py:212  >	        if vcert != "system":
cloudinit/config/cc_byobu.py:56   >	    if value == "user" or value == "system":
cloudinit/distros/freebsd.py:54   >	            "system": '--system',
cloudinit/distros/freebsd.py:66   >	        if 'no_create_home' in kwargs or 'system' in kwargs:
cloudinit/distros/netbsd.py:42   >	            "system": '--system',
cloudinit/distros/netbsd.py:54   >	        if 'no_create_home' not in kwargs or 'system' not in kwargs:
cloudinit/distros/__init__.py:419  >	            "system": '--system',
cloudinit/distros/__init__.py:467  >	        if kwargs.get('no_create_home') or kwargs.get('system'):
cloudinit/analyze/show.py:200  >	            util.system_info()['system'].lower():
cloudinit/analyze/tests/test_boot.py:13   >	                                                            'system': ''})
```

so I don't think there would be far-reaching effects of this.

Reply:
Semantically speaking, 'bsd' is not a platform, it's a family of operating system. This will be confusing, or we introduce a new 'family' key. Overall, I feel like we should just use the Python`playform` module here. For instance, we can use `platform.system` instead (https://docs.python.org/3/library/platform.html#platform.system).

Reply:
`system_info()['platform']` already uses `platform.platform` [here](https://github.com/canonical/cloud-init/blob/master/cloudinit/util.py#L654). I agree that overloading "platform" is confusing[0]. Let's set that suggestion to the side.

The root of my comment is that the string comparison here makes me uncomfortable. In the other places we're comparing against `variant`, it's always against a bounded set of options ("freebsd", "netbsd", or `cloudinit.net.sysconfig.KNOWN_DISTROS`). I'd probably prefer to test against the set of known BSDs here.

[0] Though if they can all share configuration, it's not clear to me that it isn't a platform, as people evidently can write code that targets "BSD". Regardless, using a different definition of "platform" to Python is not a winning path forward.

Reply:
family would be the best

and on Linux we can reuse it for when we can group distros - like we do in puppet!

Reply:
Thanks for your inputs. I saw we already do `'BSD' in platform.system()` above, so I assume it's acceptable. I reused this strategy.